extensions/machine-learning-services/src/packageManagement/sqlRPackageManageProvider.ts

```diff
@@ -0,0 +1,154 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+'use strict';
+
+import * as vscode from 'vscode';
+import * as azdata from 'azdata';
+import * as nbExtensionApis from '../typings/notebookServices';
+
+import { QueryRunner } from '../common/queryRunner';
+import { ApiWrapper } from '../common/apiWrapper';
+import { ProcessService } from '../common/processService';
+import { Config } from '../configurations/config';
+import { SqlPackageManageProviderBase } from './SqPackageManageProviderBase';
+
+const installMode = 'install';
+const uninstallMode = 'uninstall';
+
+/**
+ * Manage Package Provider for python packages inside SQL server databases
+ */
+export class SqlRPackageManageProvider extends SqlPackageManageProviderBase implements nbExtensionApis.IPackageManageProvider {
+
+	private _rExecutable: string;
+
+	public static ProviderId = 'sql_R';
+
+	/**
+	 * Creates new a instance
+	 */
+	constructor(
+		private _outputChannel: vscode.OutputChannel,
+		apiWrapper: ApiWrapper,
+		private _queryRunner: QueryRunner,
+		private _processService: ProcessService,
+		private _config: Config) {
+		super(apiWrapper);
+		this._rExecutable = this._config.rExecutable;
+	}
+
+	/**
+	 * Returns provider Id
+	 */
+	public get providerId(): string {
+		return SqlRPackageManageProvider.ProviderId;
+	}
+
+	/**
+	 * Returns package target
+	 */
+	public get packageTarget(): nbExtensionApis.IPackageTarget {
+		return { location: 'SQL', packageType: 'R' };
+	}
+
+	/**
+	 * Returns list of packages
+	 */
+	public async listPackages(): Promise<nbExtensionApis.IPackageDetails[]> {
+		let packages = await this._queryRunner.getRPackages(await this.getCurrentConnection());
+		if (packages) {
+			packages = packages.sort((a, b) => a.name.localeCompare(b.name));
+		} else {
+			packages = [];
+		}
+		return packages;
+	}
+
+	/**
+	 * Installs given packages
+	 * @param packages Packages to install
+	 * @param useMinVersion minimum version
+	 */
+	async installPackages(packages: nbExtensionApis.IPackageDetails[], useMinVersion: boolean): Promise<void> {
+		if (packages) {
+
+			// TODO: install package as parallel
+			for (let index = 0; index < packages.length; index++) {
+				const element = packages[index];
+				await this.executeScripts(installMode, element);
+			}
+		}
+		//TODO: use useMinVersion
+		console.log(useMinVersion);
+	}
```

Comment:
Don't really see this being useful - suggest just removing it if you're not going to use the param for anything

Reply:
I get a min version not used error and I need to have the arg because of the interface. added a todo to fix later

---
In reply to: [366616911](https://github.com/microsoft/azuredatastudio/pull/8870#discussion_r366616911) [](ancestors = 366616911)

Reply:
From the tslint rule? You can just disable that with a comment
`/* tslint:disable:no-unused-variable */`

e.g. 
https://github.com/microsoft/azuredatastudio/blob/master/extensions/big-data-cluster/src/bigDataCluster/controller/apiGenerated.ts#L20

Reply:
I know but I do need to fix this by passing the version, 

---
In reply to: [366655718](https://github.com/microsoft/azuredatastudio/pull/8870#discussion_r366655718) [](ancestors = 366655718)